src/child/src/backup.rs

```diff
@@ -0,0 +1,63 @@
+use crate::store::DATA;
+use candid::{candid_method, Decode, Encode};
+use ic_canister_backup::{logic::BACKUP, models::Chunk};
+use ic_cdk::caller;
+use ic_cdk_macros::{query, update};
+use ic_scalable_canister::store::Data;
+use shared::report_model::Report;
+
+#[update(guard = "is_owner")]
+#[candid_method(update)]
+fn backup_data() -> String {
+    DATA.with(|data| {
+        let serialized = Encode!(&*data).expect("Failed to encode data");
```

Comment:
After serializing, you could immediatly deserialize into the expected type and discard the result. A backup is then only made if it can be deserialized back into the expected type